source/features/comments-time-machine-links.tsx

```diff
@@ -112,21 +72,48 @@ async function showTimeMachineBar(): Promise<void | false> {
 	);
 }
 
-function updateMenu({delegateTarget: menu}: delegate.Event<MouseEvent, HTMLElement>): void {
-	if (menu.classList.contains('rgh-time-machine-links')) {
-		return;
-	}
+function addInlineLinks(menu: HTMLElement, timestamp: string): void {
+	const comment = menu.closest('.js-comment')!;
+	const links = select.all<HTMLAnchorElement>(`
+		[href^="${location.origin}"][href*="/blob/"]:not(.${linkifiedURLClass}),
+		[href^="${location.origin}"][href*="/tree/"]:not(.${linkifiedURLClass})
+	`, comment);
 
-	const timestamp = menu.closest('.js-comment:not(.timeline-comment-group), .js-timeline-item')!.querySelector('relative-time')!.attributes.datetime.value;
+	for (const link of links) {
+		const linkParts = link.pathname.split('/');
+		// Skip permalinks
+		if (/^[\da-f]{40}$/.test(linkParts[4])) {
+			continue;
+		}
 
-	addDropdownLink(menu, timestamp);
-	addInlineLinks(menu, timestamp);
-	menu.classList.add('rgh-time-machine-links');
+		const searchParameters = new URLSearchParams(link.search);
+		searchParameters.set('rgh-link-date', timestamp);
+		link.search = String(searchParameters);
+	}
+}
+
+function addDropdownLink(menu: HTMLElement, timestamp: string): void {
+	select('.show-more-popover', menu.parentElement!)!.append(
+		<div className="dropdown-divider"/>,
+		<a
+			href={buildRepoURL(`tree/HEAD@{${timestamp}}`)}
+			className={'dropdown-item btn-link ' + linkifiedURLClass}
+			role="menuitem"
+			title="Browse repository like it appeared on this day"
+		>
+			View repo at this time
+		</a>,
+	);
 }
 
 function init(): void {
-	// Use `click` handler instead of `toggle` because there's no easy selector just for `details`
-	delegate(document, '.js-reaction-popover-container ~ details:last-child', 'click', updateMenu);
+	const commentPopupMenus = select.all('.js-reaction-popover-container ~ details:last-child:not(.rgh-time-machine-links)');
+	for (const menu of commentPopupMenus) {
```

Comment:
Dropped the delegate in favor of a plain loop, but kept the same selector because it works well with all kinds of comments.

Not sure if we really need the deduplication class still.